app/models/krikri/provider.rb

```diff
@@ -157,7 +157,8 @@ def initialize_name
                        :fl => 'provider_name' }
       response = Krikri::SolrResponseBuilder.new(query_params).response
       return nil unless response.docs.count > 0
-      response.docs.first['provider_name'].first
+      response.docs.first['provider_name'].respond_to?(:first) ?
+        response.docs.first['provider_name'].first : nil
```

Comment:
:+1: :clap: This is great. It solves the problem--for this case--of whether `nil` really means "nothing" or "this other thing that is slightly an odd issue" in a nicely encapsulated way.